### PR TITLE
node: fix AlternatingFileWriter crash on high load

### PR DESCRIPTION
**Pull request by perf2711**

### CODE REVIEW

File: packages/node/src/BacktraceClient.ts [hunk 1]
```diff
@@ -40,6 +46,7 @@ export class BacktraceClient extends BacktraceCoreClient<BacktraceConfiguration>
             breadcrumbsManager.setStorage(
                 FileBreadcrumbsStorage.create(
                     this.sessionFiles,
+                    fileSystem,
```
Comment (konraddysput) [inline packages/node/src/BacktraceClient.ts:49]: it should be taken from clientSetup if possible
```suggestion
                    clientSetup?.fileSystem ?? fileSystem,
```

File: packages/sdk-core/tests/_mocks/fileSystem.ts [hunk 2]
```diff
@@ -2,11 +2,11 @@ import path from 'path';
 import { BacktraceFileAttachment } from '../../src/model/attachment';
 import { FileSystem } from '../../src/modules/storage/FileSystem';
 
-type MockedFileSystem = {
-    [K in keyof FileSystem]: jest.Mock<ReturnType<FileSystem[K]>, Parameters<FileSystem[K]>>;
+export type MockedFileSystem<T extends FileSystem> = {
```
Comment (konraddysput) [inline packages/sdk-core/tests/_mocks/fileSystem.ts:5]: can we fix the warning and add es-lint ignore?

File: packages/node/src/breadcrumbs/FileBreadcrumbsStorage.ts [hunk 3]
```diff
@@ -27,12 +28,16 @@ export class FileBreadcrumbsStorage implements BreadcrumbsStorage {
     constructor(
         private readonly _mainFile: string,
         private readonly _fallbackFile: string,
+        fileSystem: NodeFileSystem,
         maximumBreadcrumbs: number,
     ) {
-        this._writer = new AlternatingFileWriter(_mainFile, _fallbackFile, maximumBreadcrumbs);
+        this._writer = new AlternatingFileWriter(_mainFile, _fallbackFile, maximumBreadcrumbs, fileSystem);
```
Comment (konraddysput) [inline packages/node/src/breadcrumbs/FileBreadcrumbsStorage.ts:34]: It is not part of this pull request, however, can we adjust the code also here? The number of breadcrumbs shouldn't extend the desired number of breadcrumbs declared by the user
```suggestion
        this._writer = new AlternatingFileWriter(
            _mainFile,
            _fallbackFile,
            Math.floor(maximumBreadcrumbs / 2),
            _fileSystem,
        );
```

**Review by konraddysput — APPROVED** [review]
Consider adjusting our AlternatingFileWriter algorithms between node and react-native. I think we should see a performance improvement if we batch entries into one file write operation

File: packages/node/src/BacktraceClient.ts (hunk 1 above)
Comment (perf2711) [reply packages/node/src/BacktraceClient.ts:49]: Fixed.

File: packages/sdk-core/tests/_mocks/fileSystem.ts (hunk 2 above)
Comment (perf2711) [reply packages/sdk-core/tests/_mocks/fileSystem.ts:5]: Added.

File: packages/node/src/breadcrumbs/FileBreadcrumbsStorage.ts (hunk 3 above)
Comment (perf2711) [reply packages/node/src/breadcrumbs/FileBreadcrumbsStorage.ts]: Changed.
